normits_demand/models/notem/production_models.py

```diff
@@ -329,26 +322,19 @@ def run(self,
             #  Output some audits of what demand was before and after control
             #  By segment.
             if self.constraint_paths is not None:
-                raise NotImplemented(
-                    "No code implemented to constrain productions."
-                )
-
+                msg= "No code implemented to constrain productions"
+                raise NotImplementedError(msg)
```

Comment:
Make sure you are logging this Error as well

`self._logger.error(msg)`
`raise NotImplementedError(msg)`